### ISSUE TITLE
Document how we will bill for Lenovo A100 GPUs on OpenShift

### ISSUE THREAD

**Issue by msdisme** [nerc-project/operations#314]
As a Lenovo representative coordinating with the MOC Alliance, I need a  guide that explains the NERC billing model for Lenovo A100 GPUs on OpenShift and OpenStack. This guide will enable the team to align our internal billing processes when charging the MOC Alliance with the usage being tracked by the NERC billing system. We need to detail all aspects of GPU usage billing, excluding MIG considerations for now, as per the guidelines in issue #307, but should be higher level and not be blocked by the completion of every permutation that will be required for #307.

## OpenShift
1. Prometheus
   1. Confirm it is an A100 GPU
      1. the below command is how we currently check GPU
      ```
      $ oc get nodes -o yaml wrk-88
      ...
        nvidia.com/gpu.product: Tesla-V100-PCIE-32GB
      ...
      ```
   1. Find out how long it has been running
      1. [This method](https://github.com/CCI-MOC/openshift-usage-scripts/blob/main/openshift_metrics/utils.py#L169) calculates the duration.

## Notes
- we will need to be able to distinguish between the Lenovo A100's and any A100's which we are not charging Lenovo for

**Comment by knikolla** [comment]
I confirm the two points under Nova.

**Comment by naved001** [comment]
> Confirm it is an A100 GPU

  >  [ set to "nvidia.com/gpu_A100"](https://github.com/CCI-MOC/openshift-usage-scripts/blob/main/openshift_metrics/utils.py#L26) - @naved001 can you fill in what the variable is that we are looking for this value in?
   
   
 That was actually just a place holder. We will need to figure out what type of GPU a pod is using AFTER we come up with a plan how a pod is scheduled in the first place. 
 
 Right now, I **think** a pod just says give me a resource of type `nvidia.com/gpu` and then it gets scheduled on a GPU pod. In theory, we could write code to query openshift to check what compute host the pod was scheduled on and then query from that the type of GPU there is, but that's not what we want. We want to the pod to be explicit about the type of GPU it needs and I think that would be the place to gather the type of GPU (it could be that a pod might say "schedule me on a node with a label or annotation matching this specific GPU, something like that). 
 
 This is the "resources" a compute host advertises.
 ```
Capacity:
  cpu:                80
  ephemeral-storage:  780260332Ki
  hugepages-1Gi:      0
  hugepages-2Mi:      0
  memory:             791162636Ki
  nvidia.com/gpu:     1
  pods:               250
Allocatable:
  cpu:                76
  ephemeral-storage:  718014178957
  hugepages-1Gi:      0
  hugepages-2Mi:      0
  memory:             786865932Ki
  nvidia.com/gpu:     1
  pods:               250
  ```
  
  As you can see it only says `nvidia.com/gpu`. The node has more annotation/label that specify the type of GPU but the pod needs to be explicit about that.
 
 
 
 > Find out how long it has been running

  >   [pod_metric_dict["duration"]](https://github.com/CCI-MOC/openshift-usage-scripts/blob/main/openshift_metrics/utils.py#L270-L279) - @naved001 can you confirm?
    
  Yes. We can figure out how long a pod is running.

**Comment by naved001** [comment]
> [pod_metric_dict["duration"]](https://github.com/CCI-MOC/openshift-usage-scripts/blob/main/openshift_metrics/utils.py#L270-L279) - @naved001 can you confirm?

There's no direct metric from prometheus that tells us how long a pod has been running. Since we get a time series data from prometheus, we *calculate* the running time of a pod by looking at the timestamps of the metrics. [This method](https://github.com/CCI-MOC/openshift-usage-scripts/blob/main/openshift_metrics/utils.py#L169) calculates the duration.

The only metrics that we retrieve from Prometheus are: 
https://github.com/CCI-MOC/openshift-usage-scripts/blob/main/openshift_metrics/openshift_prometheus_metrics.py#L27-L29

**Comment by joachimweyl** [comment]
currently V100s show up like this in OpenShift
```
$ oc get nodes -o yaml wrk-88
...
  nvidia.com/gpu.product: Tesla-V100-PCIE-32GB
...
```

**Comment by joachimweyl** [comment]
Splitting this into 2 issues one for OpenStack and one for OpenShift since we already figured out OpenStack. The OpenStack issue is [here](https://github.com/nerc-project/operations/issues/347).

**Comment by msdisme** [comment]
if dependency on GPU arrival target is a future sprint.

**Comment by joachimweyl** [comment]
compute node name needed

**Comment by joachimweyl** [comment]
@naved001 what does this command return for our new Lenovo nodes?
`oc get nodes -o yaml wrk-95`

**Comment by naved001** [comment]
@joachimweyl 
```
naved@computer ~ % oc get node wrk-95 -o yaml | grep -E '\s+nvidia.com/gpu.(product|count)'
    nvidia.com/gpu.count: "4"
    nvidia.com/gpu.product: NVIDIA-A100-SXM4-40GB
  ```

**Comment by joachimweyl** [comment]
@msdisme is this a duplicate of https://github.com/nerc-project/operations/issues/307
